File: DailyChallenge/2020/March/21.py
# https://leetcode.com/explore/challenge/card/march-leetcoding-challenge-2021/590/week-3-march-15th-march-21st/3679/

class Solution(object):
    def reorderedPowerOf2(self, N):
        """
        :type N: int
        :rtype: bool
        """
        c = collections.Counter(str(N))
        return any(c == collections.Counter(str(1 << i)) for i in xrange(32))

File: DailyChallenge/2020/April/21.java
https://leetcode.com/explore/challenge/card/april-leetcoding-challenge-2021/595/week-3-april-15th-april-21st/3715/

public class Solution {
    public int minimumTotal(List<List<Integer>> triangle) {
        for (int i = triangle.size() - 2; i >= 0; --i) {
            List<Integer> cur = triangle.get(i);
            List<Integer> next = triangle.get(i+1);
            for (int j = 0; j < i + 1; ++j) {
                cur.set(j, Math.min(next.get(j), next.get(j+1)) + cur.get(j));   
            }
        }
        return triangle == null ? 0 : triangle.get(0).get(0);
    }
}
